### PR TITLE
feat: add types for module configuration

### PR DESCRIPTION
**Pull request by rchl**

### CODE REVIEW

**Comment by rchl** [comment]
@kevinmarrec maybe you want to have a look? Not necessarily from the module-specific point of view but more a generic TS types point of view.

File: types/sentry.d.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    disableClientSide?: boolean
+    disabled?: boolean
+    disableServerRelease?: boolean
+    disableServerSide?: boolean
+    dsn?: string | ''
+    initialize?: boolean
+    lazy?: boolean | LazyConfiguration
+    logMockCalls?: boolean
+    publishRelease?: boolean
+    repo?: string | ''
```
Comment (kevinmarrec) [inline types/sentry.d.ts:31]:
```suggestion
    repo?: string
```

`''` is of type `string` 

image

File: types/sentry.d.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    serverConfig?: SentryOptions
+    serverIntegrations?: IntegrationsConfiguration
+    sourceMapStyle?: WebpackOptions.Devtool
+    webpackConfig?: SentryCliPluginOptions
+}
+
+type LazyConfigurationResolved = Required<LazyConfiguration>
+type ModuleConfigurationResolved = Required<ModuleConfiguration>
```
Comment (kevinmarrec) [inline types/sentry.d.ts:39]: These types are not used ? Maybe you wanted to export them ?

Comment (rchl) [reply types/sentry.d.ts:39]: Those are internal and I'm using them through JSDoc like so: https://github.com/nuxt-community/sentry-module/blob/bc65423448b391fa54c3322b19331f429e6deeea/lib/core/hooks.js#L30

I suppose I can create them dynamically in JSDoc comments instead.

Comment (rchl) [reply types/sentry.d.ts]: I'm creating those in JSDoc now although still not sure which of the three possible options I should go with 👍 
 - declare internal here and not export
 - declare and export
 - create directly in jsdoc

Comment (kevinmarrec) [reply types/sentry.d.ts]: Didn't know it was possible to use unexported types with JSDoc ^^

If only used in internals :
 - create directly in jsdoc
 - declare and export (but not export in main types entry a.k.a. index.d.ts)

Current state looks good to me

Comment (rchl) [reply types/sentry.d.ts]: Thanks

**Pull request merged**